cedar-policy-core/src/evaluator/err.rs

```diff
@@ -44,6 +48,23 @@ impl Diagnostic for EvaluationError {
         }
     }
 
+    fn source_code(&self) -> Option<&dyn miette::SourceCode> {
+        self.source_loc
+            .as_ref()
+            .map(|loc| &loc.src as &dyn miette::SourceCode)
+            .or_else(|| self.error_kind.source_code())
+    }
+
+    fn labels(&self) -> Option<Box<dyn Iterator<Item = miette::LabeledSpan> + '_>> {
```

Comment:
We could, but I'm worried that sometimes we have the same source location both in the `source_loc` and in the `error_kind`.  Would miette handle that correctly?  or would it underline the span twice or something?  I can experiment

Reply:
Tested this and it does underline the span twice (kinda) if you put the same span in the iterator twice.  I think we should avoid doing that.

It makes sense to me to provide deference to the explicit `source_loc` in the structure if it's `Some`, with documentation that that's what's happening.

The vast majority of the `EvaluationErrorKind`s do not have a source loc field -- the only cases that do have a source loc are `InvalidRestrictedExpression` and `NonValue`, and those just because they store an `Expr` as part of the error.